### PR TITLE
Auto detect feed in common server

### PR DESCRIPTION
**Pull request by orhovy**
## Status
- [X] Ready

## Description
Added a new function auto_detect_indicator_type  for the issue: https://github.com/demisto/etc/issues/23939

## Does it break backward compatibility?
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by orhovy** [comment]
I tested the change locally, changed the cvefeed to a docker that has tldextract python module installed and ran the feed from demisto.

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 1]
```diff
@@ -2709,7 +2764,7 @@ def replace_str(src_str):
 urlRegex = r'(?:(?:https?|ftp|hxxps?):\/\/|www\[?\.\]?|ftp\[?\.\]?)(?:[-\w\d]+\[?\.\]?)+[-\w\d]+(?::\d+)?' \
            r'(?:(?:\/|\?)[-\w\d+&@#\/%=~_$?!\-:,.\(\);]*[\w\d+&@#\/%=~_$\(\);])?'
 cveRegex = r'(?i)^cve-\d{4}-([1-9]\d{4,}|\d{4})$'
-
+domainRegex = r'(?i)(?:(?:https?|ftp|hxxps?):(\/?\/?\\?\\?)|www\[?\.\]?|ftp\[?\.\]?)(?:[-\d\p{L}]+\[?\.\]?)+[-\d\p{L}]+(?::\d+)?(?:((\/)|(\*|\?))+[-\d\p{L}+&@#\/%=~_$?!\-:,.\(\);\\*–—―|]+)?|\b[A-Za-z0-9._%=+\p{L}-]+@[A-Za-z0-9\p{L}.-]+\.[A-Za-z]{2,}\b'
```
Comment (bakatzir) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:2767]: I do not think we use this eventually.

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython_test.py [hunk 2]
```diff
@@ -1768,3 +1769,45 @@ def test_append_context(mocker, context_mock, data_mock, key, expected_answer):
...
+    ('joe@gmail.com', 'Email'),
+    ('koko@demisto.com', 'Email'),
+    ('42a5e275559a1651b3df8e15d3f5912499f0f2d3d1523959c56fc5aea6371e59', 'File'),
+    ('10676cf66244cfa91567fbc1a937f4cb19438338b35b69d4bcc2cf0d3a44af5e', 'File'),
+    ('52483514f07eb14570142f6927b77deb7b4da99f', 'File'),
+    ('c8092abd8d581750c0530fa1fc8d8318', 'File'),
+    ('fe80:0000:0000:0000:91ba:7558:26d3:acde', 'IPv6'),
+    ('fd60:e22:f1b9::2', 'IPv6'),
+    ('2001:db8:0000:0000:0000:0000:0000:0000', 'IPv6'),
+    ('112.126.94.107', 'IP')
```
Comment (bakatzir) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython_test.py:1799]: maybe add a `None` check as well?

**Comment by ronykoz** [comment]
if i'm not mistaken docstring should be in another format look at the other functions in the module for reference

**Pull request closed without merging**